src/main/java/io/outfoxx/swiftpoet/CodeWriter.kt

```diff
@@ -376,7 +379,9 @@ internal class CodeWriter(
    */
   private fun resolveImport(typeName: DeclaredTypeName): String {
     val topLevelTypeName = typeName.topLevelTypeName()
-    return if (importedTypes[typeName.topLevelTypeName().simpleName]?.count() == 1 && importedTypes.values.flatMap { it }.any { it == topLevelTypeName }) {
+    return if (topLevelTypeName.moduleName.isNotEmpty() && sameModuleReferencedTypes.contains(topLevelTypeName.simpleName)) {
+      typeName.canonicalName
+    } else if (importedTypes[topLevelTypeName.simpleName]?.count() == 1 && importedTypes.values.flatMap { it }.any { it == topLevelTypeName }) {
```

Comment:
This changed from `[typeName.topLevelTypeName().simpleName]` to `[topLevelTypeName.simpleName]` did not need to calculate it again since its already stored above.